File: 222-count-complete-tree-nodes/count-complete-tree-nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var countNodes = function(root) {
    let queue = [root];
    let count = 0;

    while(queue[0]) {
        const currNode = queue.shift();
        count++;

        if(currNode.left) queue.push(currNode.left);
        if(currNode.right) queue.push(currNode.right);
    }

    console.log("count", count);
    return count;
    
};

File: 222-count-complete-tree-nodes/count-complete-tree-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def countNodes(self, root: Optional[TreeNode]) -> int:
        queue = []
        queue.append(root)

        count =  0
        while len(queue) > 0:
            qLength = len(queue)
            for i in range(qLength):
                node = queue.pop(0)
                if node: count += 1
                if node and node.left:
                    queue.append(node.left)
                if node and node.right:
                    queue.append(node.right)
        return count 